paddle/pten/core/storage.h

```diff
@@ -114,24 +113,18 @@ class TensorStorage : public Storage {
   size_t size() const noexcept override { return size_; }
 
   const Place& place() const override {
-    if (!data_ && !alloc_) {
+    if (!data_) {
       PADDLE_THROW(paddle::platform::errors::Unimplemented(
           "Unable to visit place: either data_ or alloc_ has to be initialized "
           "first."));
     }
-    if (data_) {
-      return data_->place();
-    }
-    return alloc_->place();
+    return data_->place();
   }
 
   bool OwnsMemory() const noexcept override { return true; }
-  const std::shared_ptr<Allocator>& allocator() const noexcept {
-    return alloc_;
-  }
 
  private:
-  const std::shared_ptr<Allocator> alloc_;
+  Allocator* alloc_;
```

Comment:
If I understand it correctly, the lifetime of Allocator is essentially controlled by the "Context" instead of the DenseTensor. If the Context happened to destruct its Allocator, then this Allocator* held by DenseTensor will be dangling.

Reply:
Your suggestion is correct. I will remove this member pointer. Regarding the type of the constructor remaining the same, do you think it's ok?

Reply:
yup, that should work.

Reply:
Thanks!